### ISSUE TITLE
Assignment 1

### ISSUE THREAD

**Issue by evinstk** [evinstk/coding-sandbox#1]
# Assignment 1
We have a new module called `sandbox.mushroom`. The goal is to create a classifier that can predict whether a mushroom type is edible or poisonous based on input data (the stakes are high!). Currently our classifier is not very good, with accuracy around 65%—most people wouldn't trust it enough to decide whether or not to eat a mushroom.

We'll go with a "choose your own adventure" approach here on what to do next. I have a few ideas on how we can enhance our classifier program, and I'll try to rank them from easiest to hardest. You decide how you want to proceed!

(FWIW, it is possible to get this classifier to just shy of 100% accuracy consistently.)

To contribute your enhancement, make a new branch in Git and commit your work to that branch. Then push the branch to this repository and make a pull request by selecting your feature branch and targeting the `main` branch for merge. The pull request will show a diff that highlights your changes. We will use the pull request to go through the code review process.

## Easy: More logging
It looks like logging was treated as an afterthought. Can we add some more logging to get more insight on how our program is behaving? Some things to consider:

- How long does it take to train the classifier?
- What other metrics can we log other than accuracy?
- What are the most important features that the classifier uses?

## Easy: Stratification
The program doesn't give any criteria for splitting the training set from the test set. Ideally there should be a similar distribution of features across both sets so the model can train with a variety of feature values and generalize appropriately. Can we add a way to stratify the training and test sets to accomplish this? (Read chapter 2 in Hands-On ML for a refresher.)

## Medium: Saving the model
The program trains a model and reports stats, but then it just throws the model away. Not very useful. Let's improve on that.

- Save the model. (Local storage is fine, but you could also try saving to blob storage so you can share with other developers.)
- Load a model for use on production data. (You can simulate this however you like. The point is to take a model from storage and be able to run it.)

## Medium: Unit tests
We have no unit tests to guarantee that our program runs reliably. Can we add some to improve the reliability of the system. (Let's pair on this one.)

## Hard: Integrate new data source
Our mushroom forager team has just delivered a new batch of data at path "coding_sandbox/mushroom/mushroom_extra.parquet" in our data blob storage container (see `data.py` for context). Integrate this new data source into the model training and see if it improves our metrics.
